### PR TITLE
KOGITO-6229 Add Oracle DB Support to Data Index

### PR DESCRIPTION
**Pull request by Scuilion**
Add Oracle support to the Data Index app.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
ok to test

**Comment by nmirasch** [comment]
Jenkins retest this

File: data-index/data-index-service/data-index-service-oracle/src/main/resources/application.properties [hunk 1]
```diff
@@ -33,14 +33,14 @@ mp.messaging.incoming.kogito-jobs-events.enable.auto.commit=false
...
-quarkus.datasource.username=
-quarkus.datasource.password=
-quarkus.datasource.jdbc.url=
+quarkus.datasource.username=system
+quarkus.datasource.password=test
+quarkus.datasource.jdbc.url=jdbc:oracle:thin:@localhost:1521/XEPDB1
 
 #Hibernate
 quarkus.hibernate-orm.jdbc.timezone=UTC
+quarkus.hibernate-orm.database.generation=create
```
Comment (Scuilion) [inline data-index/data-index-service/data-index-service-oracle/src/main/resources/application.properties]: It's probably not a good idea to have this here. I'll add a `%` profile before PR completion.

**Comment by radtriste** [comment]
Jenkins retest this

File: data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/model/UserTaskInstanceEntity.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    @CollectionTable(name = "tasks_potential_users", joinColumns = @JoinColumn(name = "task_id", foreignKey = @ForeignKey(name = "fk_tasks_potential_users_tasks")))
+    @Column(name = "user_id", nullable = false)
+    private Set<String> potentialUsers;
+    private String referenceName;
+    private ZonedDateTime lastUpdate;
+    private String processId;
+    private String rootProcessId;
+    private String rootProcessInstanceId;
+    @Type(type = "jsonb")
+    //@Column(columnDefinition = "jsonb")
```
Comment (nmirasch) [inline data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/model/UserTaskInstanceEntity.java]: @Scuilion could this comment be removed?

File: data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/model/UserTaskInstanceEntity.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    private ObjectNode inputs;
+    @Type(type = "jsonb")
+    //@Column(columnDefinition = "jsonb")
```
Comment (nmirasch) [inline data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/model/UserTaskInstanceEntity.java]: @Scuilion same here

File: persistence-commons/persistence-commons-oracle/src/main/java/org/kie/kogito/persistence/oracle/model/CacheEntity.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    @Id
+    @Column(nullable = false)
+    private String name;
+
+    @Id
+    @Column(nullable = false)
+    private String key;
+
+    @Type(type = "jsonb")
+    @Column(name = "json_value") //, columnDefinition = "clob")
```
Comment (nmirasch) [inline persistence-commons/persistence-commons-oracle/src/main/java/org/kie/kogito/persistence/oracle/model/CacheEntity.java]: @Scuilion could you remove this comment?

File: persistence-commons/persistence-commons-oracle/src/test/java/org/kie/kogito/persistence/schema/DDLSchemaExporter.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Environment;
+import org.hibernate.tool.hbm2ddl.SchemaExport;
+import org.hibernate.tool.schema.TargetType;
+import org.kie.kogito.persistence.oracle.model.CacheEntity;
+import org.kie.kogito.testcontainers.KogitoOracleSqlContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+//TODO: remove?
```
Comment (nmirasch) [inline persistence-commons/persistence-commons-oracle/src/test/java/org/kie/kogito/persistence/schema/DDLSchemaExporter.java]: @Scuilion I think is a good idea to keep this DLLSchemaExporter as a nice tool to have. Maybe just remove  the TODO comment would work.

File: data-index/data-index-service/data-index-service-oracle/src/test/docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      KAFKA_LISTENERS: INTERNAL://kafka:29092,EXTERNAL://kafka:9092
+      KAFKA_ADVERTISED_LISTENERS: INTERNAL://kafka:29092,EXTERNAL://localhost:9092
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: INTERNAL
+      KAFKA_AUTO_CREATE_TOPICS_ENABLE: "true"
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      LOG_DIR: "/tmp/logs"
+
+  oracle:
+    image: gvenzl/oracle-xe:21.3.0
```
Comment (nmirasch) [inline data-index/data-index-service/data-index-service-oracle/src/test/docker/docker-compose.yml]: @Scuilion here you are using image: gvenzl/oracle-xe:21.3.0 and in persistence-commons-oracle you are using gvenzl/oracle-xe:21.3.0-slim. Is there any reason for that difference?

**Review by nmirasch — COMMENTED** [review]
In general looks really good, thanks! . I just found some minor things like comments in code. From my side, I'm still trying to use this persistence in an example, but If something comes out we can handle in further PRs.

**Comment by nmirasch** [comment]
Jenkins retest this

File: data-index/data-index-service/data-index-service-oracle/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,47 @@
+quarkus.package.type=uber-jar
+
+#Data Index
+kogito.persistence.type=oracle
+kogito.data-index.domain-indexing=false
+kogito.data-index.blocking=true
+
+#Kafka
```
Comment (nmirasch) [inline data-index/data-index-service/data-index-service-oracle/src/main/resources/application.properties:8]: @Scuilion  I needed to add here kafka.bootstrap.servers=http://localhost:9092 to see this working with the example. But I guess that is something that we can add in the kogito-data-index-oracle image configuration

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Comment by cristianonicolai** [comment]
@Scuilion can you try to rebase your PR as Jenkins keeps failing.

**Review by nmirasch — APPROVED** [review]
Looks good to me and  it works fine in a real example. Good to go after achieve the jenkins green, thanks @Scuilion!

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Review by cristianonicolai — APPROVED** [review]
@Scuilion many thanks, great work!

**Comment by cristianonicolai** [comment]
@spolti fyi this needs to be follow up by https://issues.redhat.com/browse/KOGITO-6596

**Comment by radtriste** [comment]
sorry I did not have time to review this ... I will check the images PR

**Pull request merged**